ncp.c

```diff
@@ -1410,8 +1412,8 @@ wait_for_ack_of_pkt(struct conn *conn, int pknum)
       tn.tv_sec--; 
       tn.tv_usec += 1000000;
     }
-    printf("NCP %s for %p done waiting for EOF ack %#x (%s after %ld.%03d)\n", conn_thread_name(conn), conn, pknum,
-	   timedout != 0 ? "timed out" : "acked", tn.tv_sec, tn.tv_usec/1000);
+    printf("NCP %s for %p done waiting for EOF ack %#x (%s after %ld.%03ld)\n", conn_thread_name(conn), conn, pknum,
+	   timedout != 0 ? "timed out" : "acked", tn.tv_sec, (long)(tn.tv_usec/1000));
```

Comment:
tv_usec is a `long int` so %ld is more appropriate.  The cast is an extra precaution.